### DISCUSSION TITLE
[Feature request] Add an `html_url` field to the subject of a notification thread in the REST API

### DISCUSSION THREAD

**Discussion by Fs00** [community/community#37536] [Apps, API and Webhooks]
Hello everyone!

I'm here to request a small but potentially very useful addition to the [Notifications REST API endpoint for getting notification threads](https://docs.github.com/en/rest/activity/notifications#get-a-thread).
Currently, the response of that endpoint contains a `subject` object which includes an `url` field. That field is the URL to the API endpoint that can be used to retrieve the issue/PR/whatever originated the notification (e.g. https://api.github.com/repos/octokit/octokit.rb/issues/123).
The problem is that the aforementioned `url` field has some important limitations:
- if the notification originated from a discussion or from a CI run that failed, the field will be empty because there is no REST API for those. In these cases, it's impossible to understand where the notification is exactly coming from.
- I want to use the Notifications REST API to build a UI that displays a list of GitHub notifications and gives the user the ability to tap on a notification to jump to the GitHub web page for that notification.
Since the `url` field returns an API URL, I need to transform it in a website URL by manipulating it (which can be error-prone), but I can't do anything if the provided URL is empty in the cases described in the previous point.

For these reasons, I propose the addition of an `html_url` field to the `subject` of a notification thread in the API response. That field should provide the same URL I get when I click on the notification in the [GitHub notifications page](https://github.com/notifications).
Thank you! :pray:

**Comment by ossanna16** [comment]
Hi there @Fs00 and welcome to our community! Thank you for the feature request 🙂

**Reply by Fs00** [reply]
Thank you! Will the request be forwarded to the relevant teams?

**Reply by ossanna16** [reply]
Yes, the relevant teams are keeping an eye on this space.

**Comment by lordmauve** [comment]
+1 but also I would like the URL to include the `notification_referrer_id` parameter, so that visiting the URL will present the toolbar to mark the notification as done etc.
